Tools/ardupilotwaf/toolchain.py

```diff
@@ -119,6 +119,10 @@ def _filter_supported_cxx_compilers(*compilers):
 def configure(cfg):
     if cfg.env.TOOLCHAIN == 'native':
         cfg.load('compiler_cxx compiler_c')
+
+        cfg.check_cxx(header_name='cmath', features='cxx cprogram', mandatory=False)
+        cfg.write_config_header('toolchain.h')
```

Comment:
maybe I wasn't so clear... you need to use check_cxx() to pass a fragment containing the function you want to check for (even inside or outside the std namespace... depends how you will do it).  Here you are only checking if there's a cmath header, which is pointless since you are including it nonetheless